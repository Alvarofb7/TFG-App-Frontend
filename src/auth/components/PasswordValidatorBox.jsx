export const PasswordValidatorBox = ({
	lowerValid,
	upperValid,
	lengthValid,
	numberValid,
	specialCharValid,
	password,
}) => {
	return (
		<main
			className="password-validation-box"
			aria-label="password-validation-box"
			style={{
				display:
					(lengthValid &&
						lowerValid &&
						upperValid &&
						numberValid &&
						specialCharValid) ||
					password === ""
						? "none"
						: "",
			}}
		>
			<div className={lengthValid ? "validated" : "not-validated"}>
				{lengthValid ? (
					<span className="list-icon green">
						<i className="fa-solid fa-check"></i>
					</span>
				) : (
					<span className="list-icon">
						<i className="fa-solid fa-xmark"></i>
					</span>
				)}
				Debe de contener entre 6 y 20 caracteres
			</div>
			<div className={lowerValid ? "validated" : "not-validated"}>
				{lowerValid ? (
					<span className="list-icon green">
						<i className="fa-solid fa-check"></i>
					</span>
				) : (
					<span className="list-icon">
						<i className="fa-solid fa-xmark"></i>
					</span>
				)}
				Al menos una letra minúscula
			</div>
			<div className={upperValid ? "validated" : "not-validated"}>
				{upperValid ? (
					<span className="list-icon green">
						<i className="fa-solid fa-check"></i>
					</span>
				) : (
					<span className="list-icon">
						<i className="fa-solid fa-xmark"></i>
					</span>
				)}
				Al menos una letra mayúscula
			</div>
			<div className={numberValid ? "validated" : "not-validated"}>
				{numberValid ? (
					<span className="list-icon green">
						<i className="fa-solid fa-check"></i>
					</span>
				) : (
					<span className="list-icon">
						<i className="fa-solid fa-xmark"></i>
					</span>
				)}
				Al menos un número
			</div>
			<div className={specialCharValid ? "validated" : "not-validated"}>
				{specialCharValid ? (
					<span className="list-icon green">
						<i className="fa-solid fa-check"></i>
					</span>
				) : (
					<span className="list-icon">
						<i className="fa-solid fa-xmark"></i>
					</span>
				)}
				Al menos un caracter especial
			</div>
		</main>
	);
};
