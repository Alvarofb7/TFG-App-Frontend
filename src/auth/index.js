export * from "./components/PasswordValidatorBox";
export * from "./pages/LoginPage";
export * from "./layout/HomePage";
