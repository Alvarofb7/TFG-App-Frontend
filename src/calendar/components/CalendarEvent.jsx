export const CalendarEvent = ({ event }) => {
	const { title, notes } = event;

	return (
		<span>
			<strong>{title}</strong> {notes && <span>- {notes}</span>}{" "}
		</span>
	);
};
