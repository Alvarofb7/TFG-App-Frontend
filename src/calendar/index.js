
export * from "./components/CalendarEvent";
export * from "./components/CalendarModal";
export * from "./components/FabAddNew";
export * from "./components/FabDelete";
export * from "../ui/components/Navbar";


export * from "./pages/CalendarPage";