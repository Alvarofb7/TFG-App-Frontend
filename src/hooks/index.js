
export * from "./useAuthStore";
export * from "./useCalendarStore";
export * from "./useForm";
export * from "./useNoteStore";
export * from "./useUiStore";