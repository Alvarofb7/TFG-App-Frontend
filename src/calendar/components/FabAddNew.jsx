import { addHours } from "date-fns";
import { useCalendarStore, useUiStore } from "../../hooks";

export const FabAddNew = () => {

    const { openDateCalendarModal } = useUiStore();
    const { setActiveEvent } = useCalendarStore();

    const handleClick = () => {
        setActiveEvent({
            title: "",
            notes: "",
            allDay: false,
            start: new Date(),
            end: addHours(new Date(), 2),
            user: {}
        })
        openDateCalendarModal();
    }

    return (
        <button
            className="btn btn-primary fab"
            onClick={ handleClick }
        >
            <i className="fas fa-plus"></i>
        </button>
    )
}
