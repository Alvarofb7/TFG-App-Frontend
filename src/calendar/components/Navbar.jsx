import { NavLink } from "react-router-dom";
import { useAuthStore } from "../../hooks";

export const Navbar = () => {

    const { user, startLogout } = useAuthStore()

    return (
        <div className="navbar navbar-expand-md navbar-dark bg-dark mb-4 px-4">
            <span className="navbar-brand">
                <i className="fas fa-calendar-alt"></i>
                &nbsp;
                { user.name || "Sin nombre definido" }
            </span>

            <div className="navbar-collapse">
                <div className="navbar-nav">
                    <NavLink
                        className={ ({ isActive }) => `nav-link ${isActive ? "active" : ""}` }
                        to="/calendar"
                    >
                        Calendario
                    </NavLink>

                    <NavLink
                        className={ ({ isActive }) => `nav-link ${isActive ? "active" : ""}` }
                        to="/notes"
                    >
                        Notas
                    </NavLink>
                    <NavLink
                        className={ ({ isActive }) => `nav-link ${isActive ? "active" : ""}` }
                        to="/kanban"
                    >
                        Kanban
                    </NavLink>
                </div>
            </div>

            <button
                className="btn btn-outline-danger"
                onClick={ startLogout }
            >
                <i className="fas fa-sign-out-alt"></i>
                &nbsp;
                <span>Salir</span>
            </button>

        </div>
    )
}
