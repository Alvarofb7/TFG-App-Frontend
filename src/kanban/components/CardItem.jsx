import { differenceInSeconds } from "date-fns";
import { useUiStore } from "../../hooks";
import { useKanbanStore } from "../../hooks/useKanbanStore";
import { status } from "../interfaces/Status";
import { useMemo } from "react";

export const CardItem = ({ data = {}, handleDragging }) => {
	const { setActiveTask, startDeleteTask } = useKanbanStore();
	const { openKanbanModal } = useUiStore();

	const handleDragEnd = () => handleDragging(false);

	const handleDragStart = (event) => {
		event.dataTransfer.setData("text", `${data.id}`);
		handleDragging(true);
	};

	const actionOpenKanbanModal = () => {
		setActiveTask(data);
		openKanbanModal();
	};

	const handleDeleteTask = (event) => {
		event.stopPropagation();
		startDeleteTask(data.id);
	};

	const newTitle = useMemo(() => {
		return data.title.length > 20
			? data.title.substring(0, 20).concat("...")
			: data.title;
	}, [data.title]);

	return (
		<div
			className={`${
				differenceInSeconds(data.finish, new Date()) < 0
					? "card-container-expired"
					: "card-container"
			}`}
			draggable
			onDragEnd={handleDragEnd}
			onDragStart={handleDragStart}
			onClick={actionOpenKanbanModal}
		>
			<div className={`card-title ${data.status === status.done && "done"}`}>
				<h5>{newTitle}</h5>
				<button
					type="button"
					className="btn btn-danger"
					onClick={handleDeleteTask}
				>
					<i className="fa-solid fa-xmark" />
				</button>
			</div>
			{differenceInSeconds(data.finish, new Date()) < 0 && (
				<p>Esta tarea ha expirado</p>
			)}
		</div>
	);
};
