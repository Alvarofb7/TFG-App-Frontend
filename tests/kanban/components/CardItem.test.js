import { fireEvent, render } from "@testing-library/react";
import { useKanbanStore, useUiStore } from "../../../src/hooks";
import { CardItem } from "../../../src/kanban/components/CardItem";
import { taskWithId } from "../../fixtures/kanbanStates";
import { Draggable } from "react-beautiful-dnd";

jest.mock("../../../src/hooks/useKanbanStore");
jest.mock("../../../src/hooks/useUiStore");

jest.mock("react-beautiful-dnd", () => ({
	Draggable: jest.fn(({ children }) => children({})),
}));

describe("Pruebas en el componente CardItem", () => {
	const mockOpenKanbanModal = jest.fn();
	const mockStartDeleteTask = jest.fn();
	const mockSetActiveTask = jest.fn();

	beforeEach(() => {
		jest.clearAllMocks();
		jest.clearAllTimers();
	});

	test("Debe de mostrar el componente correctamente", () => {
		useUiStore.mockReturnValue({
			openKanbanModal: mockOpenKanbanModal,
		});

		useKanbanStore.mockReturnValue({
			setActiveTask: mockSetActiveTask,
			startDeleteTask: mockStartDeleteTask,
		});

		const { container } = render(<CardItem data={taskWithId} index={1} />);

		expect(container).toMatchSnapshot();
	});

	test("Debe mostrar el título correctamente", () => {
		useUiStore.mockReturnValue({
			openKanbanModal: mockOpenKanbanModal,
		});

		useKanbanStore.mockReturnValue({
			setActiveTask: mockSetActiveTask,
			startDeleteTask: mockStartDeleteTask,
		});

		const { getByText } = render(<CardItem data={taskWithId} index={1} />);
		const title = getByText("Prueba");
		expect(title).toBeInTheDocument();
	});

	test("Debe llamar a la función para abrir el modal al hacer clic en el componente", () => {
		useUiStore.mockReturnValue({
			openKanbanModal: mockOpenKanbanModal,
		});

		useKanbanStore.mockReturnValue({
			setActiveTask: mockSetActiveTask,
			startDeleteTask: mockStartDeleteTask,
		});
		const { getByText } = render(<CardItem data={taskWithId} index={1} />);
		const cardContainer = getByText("Prueba");
		fireEvent.click(cardContainer);
		expect(mockSetActiveTask).toHaveBeenCalledWith(taskWithId);
		expect(mockOpenKanbanModal).toHaveBeenCalled();
	});

	test("Debe llamar a la función para eliminar la tarea al hacer clic en el botón de eliminar", () => {
		useUiStore.mockReturnValue({
			openKanbanModal: mockOpenKanbanModal,
		});

		useKanbanStore.mockReturnValue({
			setActiveTask: mockSetActiveTask,
			startDeleteTask: mockStartDeleteTask,
		});
		const { getByLabelText } = render(<CardItem data={taskWithId} index={1} />);
		const deleteButton = getByLabelText("btn-delete-task");
		
		fireEvent.click(deleteButton);
		expect(mockStartDeleteTask).toHaveBeenCalledWith(taskWithId.id);
	});

	test("Debe renderizar el componente Draggable correctamente", () => {
		useUiStore.mockReturnValue({
			openKanbanModal: mockOpenKanbanModal,
		});

		useKanbanStore.mockReturnValue({
			setActiveTask: mockSetActiveTask,
			startDeleteTask: mockStartDeleteTask,
		});
		render(<CardItem data={taskWithId} index={1} />);
		expect(Draggable).toHaveBeenCalledWith(
			expect.objectContaining({
				draggableId: taskWithId.id,
				index: 1,
			}),
			{}
		);
	});
});
