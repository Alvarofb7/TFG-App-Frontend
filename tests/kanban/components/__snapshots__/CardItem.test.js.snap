// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente CardItem Debe de mostrar el componente correctamente 1`] = `
<div>
  <div
    class="card-container"
  >
    <div
      class="card-title false"
    >
      <h5>
        Prueba
      </h5>
      <button
        aria-label="btn-delete-task"
        class="btn btn-danger"
        type="button"
      >
        <i
          class="fa-solid fa-xmark"
        />
      </button>
    </div>
  </div>
</div>
`;
